src/DisasmViewer.cpp

```diff
@@ -281,6 +282,45 @@ void DisasmViewer::paintEvent(QPaintEvent* e)
 	visibleLines -= partialBottomLine;
 }
 
+void DisasmViewer::copyCodeToClipboard() const
+{
+    QClipboard *clipboard = QApplication::clipboard();
+	QString buffer;
+	
+	// calculate lines while writing
+	int currentLine = 0;
+	int y = frameT;
+
+	const DisasmRow* row;
+	bool displayDisasm = memory != nullptr && isEnabled();
+
+	while (y < height() - frameB) {
+		// fetch the data for this line
+		row = displayDisasm
+		    ? &disasmLines[disasmTopLine + currentLine]
+		    : &DISABLED_ROW;
+		int h;
+		switch (row->rowType) {
+			case DisasmRow::INSTRUCTION:
+				buffer += QString("%1%2\n")
+						.arg(row->instr.substr(0, 7).c_str())
+						.arg(row->instr.substr(7   ).c_str());
+				h = codeFontHeight;
+				break;
+			case DisasmRow::LABEL:
+				buffer += QString("%1:\n").arg(row->instr.c_str());
+				h = labelFontHeight;
+				break;
+		}
+		// next line
+		y += h;
+		++currentLine;
+		// check for end of data
+		if (disasmTopLine + currentLine == int(disasmLines.size())) break;
```

Comment:
This is correct, but it took me a moment to understand the logic. I wonder if it becomes clearer if you replace the while loop (line 297) with a for loop:
```
if (!memory || !isEnabled()) return;
...
int maxY = height() - frameB;
int maxLine = int(disasmLines.size());
for (int currentLine = disasmTopLine, y = frameT;
     (y < maxY) && (currentLine < maxLine);
     ++currentLine) {
    ...
    const DisasmRow* row = disasmLines[currentLine];
    ...
}
```